File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    answer = 0
    q = []
    sequence = []
    idx = 0
    for i, p in enumerate(priorities):
        q.append([i,p])
    
    while q:
        flag = True
        idx, p = q.pop(0)
        for i,x in q:
            if x > p:
                q.append([idx,p])
                flag = False
                break
        if flag:
            sequence.append(idx)
    return sequence.index(location)+1

File: 프로그래머스/2/42587. 프로세스/프로세스.cpp
#include <string>
#include <vector>
#include <queue>
#include <algorithm>
#include <iostream>

using namespace std;

int solution(vector<int> priorities, int location) {
    int answer = 0;
    queue<int> nums;
    vector<int> v;
    
    for(int i = 0; i < priorities.size(); i++){
        nums.push(i);
    }
    
    while(!nums.empty()){
        int n = nums.front();
        if(priorities[n] < *max_element(priorities.begin(), priorities.end())){
            nums.pop();
            nums.push(n);
        }
        else{
            priorities[n] = 0;
            nums.pop();
            v.push_back(n);
        }
    }
    return find(v.begin(), v.end(), location)-v.begin()+1;
    return answer;
}